crates/starknet/src/starknet/mod.rs

```diff
@@ -418,6 +527,23 @@ impl Starknet {
         add_invoke_transaction::add_invoke_transaction(self, invoke_transaction)
     }
 
+    // save starknet transactions to file
+    pub fn dump_transactions(&self) {
+        // TODO: attach in OnTransaction mode or dump all??
```

Comment:
@FabijanC Fabio in the old devnet we used the same function for the dump on the transaction and on exit, so I left this comment. There was a reason to do so?

Reply:
Also as follow up: what is the point of having `OnTransaction`  if we redump all of them?

Reply:
Regarding why the same dumping function was used for on-transaction and on-exit dumping: I did it because it was the simplest way of dumping. I agree that appending transactions would be more efficient, but we probably won't be doing such big interventions in the old devnet.

Reply:
This will be solved in separate issue

Reply:
@taco-paco and about "OnTransaction if we redump all of them" -> one of the reasons is an unexpected crash, in that case gently showdown might not be detected and data can be lost

Reply:
Don't see this problem. Data shall be flushed on every dump call. So if we unexpectedly crush we will get the same state as with current way

Reply:
There's [SerializeSeq trait](https://docs.rs/serde/latest/serde/ser/trait.SerializeSeq.html) that can help with that. Some modifications has to made tho to always get valid json